### PR TITLE
Feature/join product details into packages

### PR DESCRIPTION
**Pull request by stevenbister**
### 🧹 Please check off the following before submitting:

- [x] Close issues
- [x] Request reviewers
- [x] Assign yourself
- [x] Add labels (e.g. `build:osc-ecommerce`)
- [x] Add project
- [x] Set project status
- [x] Add milestone

---

## ☑ List all github issues that this PR will close:

- Closes https://github.com/Open-Study-College/osc/issues/978
- Closes https://github.com/Open-Study-College/osc/issues/977

## 📝 Description

Updates the Tabs component to pull the content from the package course components. To do this I'm getting the metafield from Shopify, and passing the items through to our Sanity query and grabbing the data in there.
This then gets passed into our tabs using `useLoaderData` and mapped/filtered through the object to get what we need.

In order for the data to show the tab needs to have some sort of content in there already to generate it. An empty content module is fine. Any content in that module will be replaced with the merged content.

I am basing the tab that it paces the data in on the tab title, not sure if this is necessarily the most secure way of doing it but I gather that the title isn't going to change now, as there's so many courses to update, and I can't think of a better way to match the tabs across the package and component courses.

I've also had to update the test now we're using `useLoaderData` as `MemoryRouter` doesn't work. https://reactrouter.com/en/6.14.2/routers/create-memory-router

## ⛳️ Current behaviour (updates)

N/a

## 🚀 New behaviour

- Adds metafields to Shopify product query
- Adds components to Sanity product query
- Pulls content from the package components and adds it to the package course

## 💣 Is this a breaking change (Yes/No): No

> Please describe any breaking changes.

## 🧾 Final Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible cross-device impacts (iOS, Android)
- [ ] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [ ] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.

## 📝 Additional Information

We only have so many packages available in the dev store so this is on the you'll wanna look at/test https://pr-1413-osc-ecommerce.fly.dev/courses/aat-accounting-level-2-and-3

### CODE REVIEW

File: packages/osc-ecommerce/app/components/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -89,8 +97,63 @@ interface TabContentProps {
...
 
+interface PackageCourse {
+    title?: string;
+    slug?: string;
+}
+
 const ModuleContent = (props: TabContentProps) => {
     const { tabItem } = props;
+    const { page } = useLoaderData<typeof loader>();
+    const packageCourses: PackageCourse[] | undefined = page?.packageComponents?.map((item) => ({
```
Comment (mattwendzina) [inline packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:108]: Feel like the title could be more specific as it's a bit vague atm. I quite like say it as it is e.g. `packageCoursesSlugAndTitle` but appreciate that's long lol, maybe `packageCoursesDetails`. 
I think it just looks a bit clearer overall then, because you have for example, `packageCoursesDetails` and `packageCourseContent`.

Not really precious about this, but it just took me a bit of time to piece together exactly what was happening and find it a bit easier to read when things are named more specifically.

File: packages/osc-ecommerce/app/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -89,8 +97,63 @@ interface TabContentProps {
...
+    const packageCourseContent = useMemo(
+        () =>
+            getTabContentFromPackageComponents({
+                packageComponents: page?.packageComponents,
+                targetTab: 'course content',
+            }),
+        [page?.packageComponents]
+    );
+
+    const examDetailsContent = useMemo(
```
Comment (mattwendzina) [inline packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:122]: This is specific to the package courses so could have that in the title, e.g. `packageCourseExamDetailsContent`... again maybe a bit wordy? 😅

File: packages/osc-ecommerce/app/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -89,8 +97,63 @@ interface TabContentProps {
...
+                packageComponents: page?.packageComponents,
+                targetTab: 'exam details',
+            }),
+        [page?.packageComponents]
+    );
+
+    const hasPackageCourseContent = packageCourseContent && packageCourseContent.length > 0;
+    const hasExamDetailsContent = examDetailsContent && examDetailsContent.length > 0;
+
+    if (hasPackageCourseContent && tabItem?.title.toLowerCase() === 'course content') {
```
Comment (mattwendzina) [inline packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:134]: Could the `tabitem?title...` check be included in the above check (line 131) so then this line would just be:
`if(hasPackageCourseContent){}...`

File: packages/osc-ecommerce/app/components/Tabs/Tabs.tsx [hunk 4]
```diff
@@ -89,8 +97,63 @@ interface TabContentProps {
...
+        return packageCourseContent?.map((item, i) => (
+            <PackageContent
+                key={`${item}-${i}`}
+                packageCourse={packageCourses ? packageCourses[i] : undefined}
+                packageCourseContent={item}
+                filter="module.accordion"
```
Comment (mattwendzina) [inline packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:140]: Would `moduleType` be a clearer prop name than `filter`?

File: packages/osc-ecommerce/app/components/Tabs/Tabs.tsx [hunk 5]
```diff
@@ -89,8 +97,63 @@ interface TabContentProps {
...
+                className={i !== 0 ? 'u-mt-2xl' : ''}
+            />
+        ));
+    }
+
+    if (hasExamDetailsContent && tabItem?.title.toLowerCase() === 'exam details') {
```
Comment (mattwendzina) [inline packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:146]: Same comment as line 134

File: packages/osc-ecommerce/app/components/Tabs/Tabs.tsx [hunk 6]
```diff
@@ -150,3 +213,54 @@ const FaqsContent = (props: TabContentProps) => {
...
 };
+
+interface PackageContentProps {
+    packageCourse?: PackageCourse;
+    packageCourseContent: module[] | undefined;
+    filter: string;
+    className: string;
+}
+
+const PackageContent = ({
```
Comment (mattwendzina) [inline packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:224]: Could this be called `PackageContentModule`? It ultimately returns a module and makes a bit more sense in the context of the parent component being called `ModuleContent`.

**Review by mattwendzina — CHANGES_REQUESTED** [review]
Looks good mate. Made a few comments on the `Tabs.tsx` mainly around it perhaps just being a bit clearer in certain places. I guess a lot of this is opinion and based on how quickly I could understand what was going on so not too precious about most of these comments and appreciate some of my naming suggestions would perhaps make things very words 😅

File: packages/osc-ecommerce/app/components/Tabs/Tabs.tsx (hunk 3 above)
Comment (stevenbister) [reply packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:134]: Could do I suppose. I wanted to keep them separate though as the `hasPackageCourseContent` is just checking the content part which is separate from the title

Comment (mattwendzina) [reply packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:134]: Fair enough. I just thought it looked a bit cleaner to do all checks in one place rather than splitting it up. I guess you could add `andTitle` to the end, but as with many of my suggestions, makes it more wordy lol. 
Happy for it to just stay as is if you'd prefer.

File: packages/osc-ecommerce/app/components/Tabs/Tabs.tsx [hunk 7]
```diff
@@ -89,8 +97,63 @@ interface TabContentProps {
...
+
 const ModuleContent = (props: TabContentProps) => {
     const { tabItem } = props;
+    const { page } = useLoaderData<typeof loader>();
+    const packageCourses: PackageCourse[] | undefined = page?.packageComponents?.map((item) => ({
+        title: item.store.title,
+        slug: item.store.slug?.current,
+    }));
+
+    const packageCourseContent = useMemo(
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-ecommerce/app/components/Tabs/Tabs.tsx:113]: Oh sorry my bad! I miss understood what you meant 😅 Yeah should be plural lol

**Comment by mattwendzina** [comment]
@hi-im-si  - As mentioned I've added on additional commit to cascade renaming of 'packageCourses' to 'packageCoursesDetails` into the interface and props for `<PackageContentModule />`. Other than this this PR had the thumbs up from me 👍

**Review by mattwendzina — APPROVED** [review]
LGTM
